src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -94,25 +128,211 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly)
+            }
+        }
+        
+        private void OutputFlags(NodeFactory factory, ref ObjectDataBuilder objData)
+        {
+            // Todo: RelatedTypeViaIATFlag when we support cross-module EETypes
+            // Todo: GenericVarianceFlag when we support variance
+            // Todo: OptionalFieldsFlag
+            // Todo: Generic Type Definition EETypes
+            
+            UInt16 flags = (UInt16)Kinds.CanonicalEEType;
+            
+            if (_type.IsArray || _type.IsPointer)
+            {
+                flags = (UInt16)Kinds.ParameterizedEEType;
+            }
+            
+            if (_type.IsValueType)
+            {
+                flags |= (UInt16)Flags.ValueTypeFlag;
+            }
+
+            if (_type.HasFinalizer)
+            {
+                flags |= (UInt16)Flags.HasFinalizerFlag;
+            }
+            
+            if (_type is MetadataType && ((MetadataType)_type).ContainsPointers)
+            {
+                flags |= (UInt16)Flags.HasPointersFlag;
+            }
+            else if (_type.IsArray)
+            {
+                ArrayType arrayType = _type as ArrayType;
+                if ((arrayType.ElementType.IsValueType && ((DefType)arrayType.ElementType).ContainsPointers) ||
+                    !arrayType.ElementType.IsValueType)
+                {
+                    flags |= (UInt16)Flags.HasPointersFlag;
+                }
+            }
+            
+            if (_type.IsInterface)
+            {
+                flags |= (UInt16)Flags.IsInterfaceFlag;
+            }
+
+            if (_type.HasInstantiation)
+            {
+                flags |= (UInt16)Flags.IsGenericFlag;
+            }
+
+            int corElementType = 0;
+            
+            // The top 5 bits of flags are used to convey enum underlying type, primitive type, or mark the type as being System.Array
+            if (_type.IsEnum)
+            {
+                TypeDesc underlyingType = _type.UnderlyingType;
+                Debug.Assert(TypeFlags.SByte <= underlyingType.Category && underlyingType.Category <= TypeFlags.UInt64);
+                corElementType = TypeFlagsToCorElementType(underlyingType);
             }
+            else if (IsWellKnownPrimitiveType(_type))
+            {
+                corElementType = TypeFlagsToCorElementType(_type);
+            }
+            else if (_type.IsArray)
+            {
+                corElementType = 0x14; // ELEMENT_TYPE_ARRAY
+            }
+            
+            if (corElementType > 0)
+            {
+                flags |= (UInt16)(corElementType << (UInt16)Flags.CorElementTypeShift);
+            }
+
+            objData.EmitShort((short)flags);
+        }
+
+        private bool IsWellKnownPrimitiveType(TypeDesc type)
+        {
+            return type.Category >= TypeFlags.Boolean && type.Category <= TypeFlags.Double;
+        }
 
+        private int TypeFlagsToCorElementType(TypeDesc type)
```

Comment:
Better name for this maybe something like `ComputeRhCorElementType`. Also, it can be static method.